src/context/authContext.jsx

```diff
@@ -0,0 +1,58 @@
+import React, { createContext, useContext, useEffect, useReducer } from "react";
+import { authReducer } from "../reducers/index";
+import { useNavigate } from "react-router-dom";
+import { actionTypes } from "../reducers/index";
+import axios from "axios";
+const authContext = createContext({});
+
+const useAuth = () => useContext(authContext);
+
+const { SIGN_IN, SIGN_OUT } = actionTypes;
+
+function AuthProvider({ children }) {
+  const navigate = useNavigate();
+  const [authState, authDispatch] = useReducer(authReducer, {
+    token: localStorage.getItem("token") || "",
+  });
+
+  const signUp = async (input) => {
+    try {
+      const res = await axios.post("/api/auth/signup", input);
+
+      if (res.status === 201) {
+        navigate("/signin");
+      }
```

Comment:
After signup u get a token using that u can directly sign in the user , there is no need for user to go to sign in page after sign up again, adding this here in case u don't want this feature